### PR TITLE
Fix for path traversal when allowing recursive mounts

### PR DESCRIPTION
**Pull request by nenne70**

### CODE REVIEW

**Comment by juliengk** [comment]
what's wrong with the shebang?

I looked at the docker script in the image and it seems good to me

```bash
#!/bin/bash

true
```

**Comment by nenne70** [comment]
The travis check fails on it. And if I executed the test manually in bash it also fails. The updated Dockerfile now has a working echo -e line. But the travis check still fails. Now I made a fix for /scripts/build which complains when doing a make. Not sure if it is related or not

**Comment by juliengk** [comment]
ok i just realised about travis failing and those commits might related. Don't worry about travis for now. I will look at your change for the path this weekend.

**Comment by juliengk** [comment]
could you please delete the last 3 commits that are not related to your initial PR.

**Comment by dantremblay** [comment]
Hi @juliengk , any more comments with this PR?  Please let us know.

**Comment by nenne70** [comment]
Is there something more you wish me to do on this, should I modify it in someway?

File: docker/allow/container.go [hunk 1]
```diff
@@ -48,7 +49,18 @@ func ContainerCreate(req authorization.Request, config *types.Config) *types.All
 	if len(cc.HostConfig.Binds) > 0 {
 		for _, b := range cc.HostConfig.Binds {
 			vol := strings.Split(b, ":")
-
+			
+			vol[0], _ = filepath.Abs(vol[0])
```
Comment (juliengk) [inline docker/allow/container.go]: executing this function will match directories on the file system and the plugin might be running in a container and i'm not sure that will be valid.

File: docker/allow/container.go [hunk 2]
```diff
@@ -407,6 +419,21 @@ func GetPortBindingString(pb *nat.PortBinding) string {
...
 
+func AllowVolumePath(vol string) bool {
+	if strings.Contains(vol, "../") {
+		return false
+	}
+	if strings.Contains(vol, "/..") {
+		return false
+	}
+
+	volevaluated, _ := filepath.EvalSymlinks(vol)
```
Comment (juliengk) [inline docker/allow/container.go]: executing this function will match directories on the file system and the plugin might be running in a container and i'm not sure that will be valid.

**Comment by juliengk** [comment]
here is my PR #40 for your prob reusing your code

**Pull request closed without merging**